### PR TITLE
fix show_labels and show_conf

### LINKED ISSUE #2046: How to not display labels and conf ?
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I would not to display labels and conf when predict images, but I failed. Can someone solve this problem ?
```python
import torch
from ultralytics import YOLO

# 加载模型
model = YOLO("runs/detect/train4/weights/best.pt")

# 预测结果
with torch.no_grad():
    results = model.predict(task="detect", source="/mnt/datasets/rebar_bale_detection/images/val", max_det=1000, conf=0.55, show_labels=False, show_conf=False, save=True, device="0")  # predict on a Image，最大检测数修改为1000，不显示标签和置信度，保存检测结果图像
```

result:
下载 (4)

### Additional

_No response_

### PR DESCRIPTION
**Pull request by AyushExel**
Fixes https://github.com/ultralytics/ultralytics/issues/2046
I think this was introduced while removing redundancy of plot() funtion from both results.py and predict.py. 
btw show_label=False should hide everything, right? This is how it was before I think

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Enhanced visualization options for object detection results in YOLO predictor.

### 📊 Key Changes
- Introduced additional plotting arguments to the predictor, including confidence scores (`conf`) and class labels (`labels`).
- Adjusted the argument handling to accommodate these new visualization options.

### 🎯 Purpose & Impact
- **Purpose:** This update is designed to provide users with more control over the visualization of detection results. By adding options to display confidence scores and class labels, users can now get a clearer and more informative visual output.
- **Impact:** Users who wish to see the confidence scores and labels directly in the output images can benefit from this change, especially for tasks that require result interpretation or presentation. It enhances the usability of the model's predictions for various applications.

### CODE REVIEW

**Comment by glenn-jocher** [comment]
@AyushExel looks good!
